x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/RetrySettings.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.retry;
+
+import org.elasticsearch.core.TimeValue;
+
+public class RetrySettings {
```

Comment:
Yeah good point. I think the reason I didn't use a record initially is because I was planning on passing in `ClusterService` so that these settings can be dynamically updatable. I think that'll require the object to be mutable which I don't think will work with a record right?

Reply:
I was running into some testing issues so I went ahead and made this accept `Settings`. I'll add the `ClusterService` in a follow up PR.